src/osdc/Objecter.cc

```diff
@@ -2764,9 +2764,9 @@ int Objecter::_calc_target(op_target_t *t, Connection *con, bool any_change)
   pg_mapping_t pg_mapping;
   pg_mapping.epoch = osdmap->get_epoch();
   if (lookup_pg_mapping(actual_pgid, &pg_mapping)) {
-    up = pg_mapping.up;
+    up = std::move(pg_mapping.up);
```

Comment:
Have you considered changing `lookup_pg_mapping()` to take pointers to `up`, `up_primary`, `acting` and `acting_primary` (i.e. same as `pg_to_up_acting_osds()`)?  A copy+move is still worse than a plain copy.